### DISCUSSION TITLE
Function Graph: C-Code feature instead of just Assembly

### DISCUSSION THREAD

**Discussion by NervousNullPtr** [NationalSecurityAgency/ghidra#3296] [Ideas]
I think it's pretty slef explanatory. It would be very neat if the function graph would also be able to display C-Code (Something like Binary Ninja) rather than just the raw assembly. This would make it easier to analyze programs.

**Comment by bdemick** [comment]
Similarly, it would be awesome to display high-level PCode in a graph view, exposing the SSA-ness of it as well. I've made some utility plugins for this but haven't had the time to put them into a graph view.

**Reply by Talos878** [reply]
Will you point me to these plugins? And are they using higher level pcode from the decompiler?

**Reply by bdemick** [reply]
Haven't had time to clean them up for release, unfortunately. Hopefully soon I'll have some bandwidth for this. The tooling you to select what level of PCode you want to see, currently just dumps them out as text in a slightly pretty-fied way that displays the varnode information explicitly.

**Comment by ryanmkurtz** [comment]
You might want to make this into an issue that we can mark as an enhancement.
